Added csv_to_json.js (#5) concerns #3

toyamarodrigo opened the pull request:
hey @thealphadollar! does this script meet the requirements for #3 ?
npm packages I used:
`prompt-sync` for asking path to CSV file and `csvtojson`.

to run the script `node csv_to_json.js`

toyamarodrigo committed:
csv_to_json.js

toyamarodrigo committed:
git ignore .vscode node_modules, added package-lock.json

toyamarodrigo committed:
Merge

toyamarodrigo commented:
the output folder should be `vanilla-i18n` like the .py script?

thealphadollar commented on csv_to_vanilla-i18n/csv_to_json.js:
```diff
@@ -0,0 +1,31 @@
+const fs = require('fs');
+const csv = require('csvtojson');
+const prompt = require('prompt-sync')();
```
Is there some way by which we can avoid using dependencies? I want to keep the overhead to minimum since this script is very crucial and needs to be run by developers - without installing anything extra.

I found something which may help us, https://nodejs.org/en/knowledge/command-line/how-to-prompt-for-command-line-input/

thealphadollar reviewed (commented):
Hey @toyamarodrigo Thank you very much for investing time in contributing to the project. The output files look really neat, thanks for the great work. Here are few suggestions I would have so that it is uniform with the project and thereafter we can merge the PR.

1. No dependency (look for comment on lines in file `cav_to_json.js`)
2. The output files should be in the folder `vanilla-i18n`
3. The output files have a weird naming associated with them. For instance when I ran on the `example.csv` file, one of the output had the name `'français'$'\r''.json'`. Can you please see how this can be rectified?
4. Add minimal instructions in the readme.md inside `csv_to_vanilla-i18n` folder on how to run the script.

toyamarodrigo committed:
Removed 'csvtojson' 'prompt-sync' dependencies. Now using NodeJS's built in modules (fs, path, readline). Output files location now is 'vanilla-i18n'. Fixed weird naming in json file after running the script.

toyamarodrigo committed:
Fix typo dirPath

toyamarodrigo commented:
@thealphadollar I made some changes!

- Removed `csvtojson` and `prompt-sync` dependencies. Now is working with `fs` `path` and `readline` Node's built in modules.
- Changed the output location of json files to the folder `vanilla-i18n`
- Fixed that weird naming after executing the script (or at least it works on mine 😆)
- Added some loggings like the Python script (Debug and Info)

Let me know If everything works fine and then I will start with the Readme file!

thealphadollar commented on csv_to_vanilla-i18n/csv_to_json.js:
```diff
@@ -0,0 +1,82 @@
+  input: process.stdin,
+  output: process.stdout,
+});
+const dirPath = path.join(__dirname, '/vanilla-18n');
```
```suggestion
const dirPath = path.join(__dirname, '/vanilla-i18n');
```

thealphadollar reviewed (approved):
This looks awesome, thanks for considering my suggestions and delivering it with more awesomeness.

This is ready to merge with a few changes:

- [ ] Remove `package-lock.json`
- [ ] Update readme with minimal instructions
- [ ] Correct folder name

toyamarodrigo committed:
Update csv_to_vanilla-i18n/csv_to_json.js

Co-authored-by: Shivam Kumar Jha <shivam.cs.iit.kgp@gmail.com>

toyamarodrigo committed:
Removed package-lock, Added README for Node script.

toyamarodrigo commented:
> This looks awesome, thanks for considering my suggestions and delivering it with more awesomeness.
> 
> This is ready to merge with a few changes:
> 
> * [x]  Remove `package-lock.json`
> * [x]  Update readme with minimal instructions
> * [x]  Correct folder name

All changes made! Please check them out 😄
I splited the README file in two sections, one with Python script and other with Node script.

And also I wanted to know if this counts as a hacktoberfest PR?

thealphadollar reviewed (approved):
LGTM! @toyamarodrigo Thank you very much for your contributions - the JS script you've created is very neat and to the point :100: Looking forward to more interactions and contributions from you.

> I splited the README file in two sections, one with Python script and other with Node script.

This was a really good idea, thanks for the proactive action.

> And also I wanted to know if this counts as a hacktoberfest PR?

Yes, this counts as a hacktoberfest PR. :partying_face:
